### DISCUSSION TITLE
Current price only sometimes appears in current price query

### DISCUSSION THREAD

**Discussion by raaahbin** [amberelectric/public-api#91] [General]
I'm using a shell script to feed current price info into openHAB via a JSONPATH transformation, but I seem to have uncovered some inconsistent behaviour from the API - perhaps someone else can clarify what is causing this! Here's the CURL command I'm using (with API key and site ID scrubbed)

```
curl -X 'GET' \
  'https://api.amber.com.au/v1/sites/_____/prices/current?next=1&previous=0&resolution=30' \
  -H 'accept: application/json' \
  -H 'Authorization: Bearer _____'

```

Sometimes - and I haven't been able to figure out when it does and doesn't happen - that will return a JSON block that begins

`"type": "CurrentInterval",`, followed by a block that begins `"type": "ForecastInterval",`

however, at other times, it includes only the block that begins

`"type": "ForecastInterval",`

Unfortunately, dumping JSON into openHAB through a shell script is a bit limited in its ability to parse, and will automatically return the first match for the label - on the bright side, that means I get data for "perKWH" every time I run the query, but on the down side it seems 50/50 whether I'm getting the actual current price or just the first forecast price after the present.

Note that if I change the URL query string to `...next=1&previous=1...` then I _always_ get a return block for the CurrentInterval... but since the first returned block is the first historical price, that (as described above) doesn't readily allow me to get the current price out of the return string.

Ideally, what I'd *really* like to be able to do is to make the query string include `next=0&previous=0`, so that I *only* get the current price... but that seems not to return anything.

In summary:
next=1, previous=1 always returns "ActualInterval", "CurrentInterval" and "ForecastInterval" in that order
next=0, previous=1 always returns "ActualInterval" and "CurrentInterval" in that order
next=1, previous=0 gives _either_ "CurrentInterval" and "ForecastInterval", or just "ForecastInterval", with no clear pattern
next=0, previous=0 seems to return nothing (just []), although I haven't tested that enough to see whether it returns "CurrentInterval" when next=1, previous=0 includes it.

I don't think this is the intended behaviour for the API, so hopefully this is something that can be fixed!

EDIT to add: when "next=1, previous=0" returns "CurrentInterval" as well as "ForecastInterval", I get exactly the behaviour I want from next=0, previous=0 (i.e. _only_ "CurrentInterval").

query:
`https://api.amber.com.au/v1/sites/________/prices/current?next=0&previous=0&resolution=30`

response:
```
[
   {
     "type": "CurrentInterval",
     "date": "2022-01-26",
     "duration": 30,
     "startTime": "2022-01-25T16:00:01Z",
     "endTime": "2022-01-25T16:30:00Z",
     "nemTime": "2022-01-26T02:30:00+10:00",
     "perKwh": 27.06596,
     "renewables": 19.594,
     "spotPerKwh": 9.58813,
     "channelType": "general",
     "spikeStatus": "none",
     "estimate": true
   }
 ]
```

**Comment by madpilot** [comment]
Can you try something for me?

The prev and next params are actually optional, so rather than setting them to 0, try omitting them and see if the results you get are consistent with what you expect.

You should always get a current interval.

If prev is 1 you should also get one actual.

If next is 1 you should always get one forecast.

**Reply by raaahbin** [reply]
That works! I alternated between that and next=0, previous=0, and omitting them consistently returned the CurrentInterval while 0/0 returned [].

Thanks!
